server/src/main/java/org/solarbank/server/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,60 @@
+package org.solarbank.server;
+
+import javax.money.UnknownCurrencyException;
+import org.solarbank.server.Messages;
+import org.solarbank.server.dto.ErrorResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class ControllerExceptionHandler {
+
+    @ExceptionHandler
+    @ResponseStatus
+    public ResponseEntity<ErrorResponse> handleUnknownCurrencyException(UnknownCurrencyException ex) {
```

Comment:
added this to handle UnkownCurrencyException thrown when an bad currency is input, but it gets overridden by the DefaultExceptionHandler.

Reply:
See my comment about validating the currency code in `EnergyTariff`. 

If do option 1 then you can remove this method as the error should go through you validation handler. If you do option 2, then just throw exception type from the controller layer where you validate the currency code.